llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -58919,6 +58919,12 @@ bool X86TargetLowering::IsDesirableToPromoteOp(SDValue Op, EVT &PVT) const {
     if (IsFoldableAtomicRMW(N0, Op) ||
         (Commute && IsFoldableAtomicRMW(N1, Op)))
       return false;
+    // When ZU is enabled, we prefer to not promote for MUL by a constant,
+    // since a 16b imulzu will not incur partial-write stalls, and may be
+    // able to fold away a zero-extend of the 16b result.
+    if (Subtarget.hasZU() && Op.getOpcode() == ISD::MUL &&
```

Comment:
I did not see any benefit to adding another fallthrough, since it would either require introducing a scope and duplicating the two getOperand() calls, or calling getOperand directly in the cast.

Reply:
Added this in the latest revision.